apps/aehyperchains/src/aehc_connector.erl

```diff
@@ -1,36 +1,98 @@
 -module(aehc_connector).
 
--export([broadcast/4, verify/2, subscribe/2]).
+-export([send_tx/1, get_block/1]).
 
--type tx_hash() :: binary().
--type tx_payload() :: binary().
+-export([tx/3, header/2, block/2]).
 
--type account() :: <<_:256>>.
+-export([publish/2, subscribe/1]).
 
 -type connector() :: atom().
 
-%% TODO
--callback broadcast(From::account(), TxPayload::tx_payload(), To::account()) -> 
-    tx_hash().
+-callback send_tx(Tx::aetx:tx()) -> binary().
 
--callback verify(TxHash::tx_hash()) -> 
-    boolean().
+-callback get_block(Num::integer()) -> block().
 
--callback subscribe(Module::atom()) -> ok.
+-export_type([tx/0, header/0, block/0]).
 
--spec broadcast(Con::connector(), From::account(), TxPayload::tx_payload(), To::account()) -> 
-                       tx_hash().
-broadcast(Con, From, TxPayload, To) ->
-    %% Additional params (like value, fee) are parent related and can be passed via appropriate config;
-    Res = Con:broadcast(From, TxPayload, To), true = is_binary(Res),
-    Res.
+-spec connector() -> connector().
+connector() ->
+    ok.
 
--spec verify(Con::connector(), TxHash::tx_hash()) -> 
-                    boolean().
-verify(Con, TxHash) ->
-    Res = Con:verify(TxHash), true = is_boolean(Res), 
-    Res.
+%%%===================================================================
+%%%  Parent chain simplified proto
+%%%===================================================================
 
--spec subscribe(Con::connector(), Module::atom()) -> ok.
-subscribe(Con, Module) ->
-    ok = Con:subscribe(Module).
+-record(tx, {
+             sender :: binary(),
+             recipient :: binary(),
+             payload :: binary()
+            }).
+
+-type tx() :: #tx{}.
+
+-record(header, {
+                 hash :: binary(),
+                 number = 0 :: integer()
+                }).
+
+-type header() :: #header{}.
+
+-record(block, {
+                header :: header(),
+                txs :: [tx()]
+               }).
+
+-type block() :: #block{}.
+
+-spec tx(Sender::binary(), Recipient::binary(), Payload::binary()) -> tx().
+tx(Sender, Recipient, Payload) when 
+      is_binary(Sender), is_binary(Recipient), is_binary(Payload) ->
+    #tx{ sender=Sender, recipient=Recipient, payload=Payload }.
+
+-spec header(Hash::binary(), Num::integer()) -> header().
+header(Hash, Num) when 
+      is_binary(Hash), is_integer(Num) ->
+    #header{ hash=Hash, number=Num }.
+
+-spec block(Header::header(), Txs::[tx()]) -> block().
+block(Header, Txs) when
+      is_record(Header, header), is_list(Txs) ->
+    #block{ header=Header, txs=Txs }.
+
+%%%===================================================================
+%%%  Parent chain interface
+%%%===================================================================
+
+-spec send_tx(Tx::aetx:tx()) ->
+                    {ok, TxHash::binary()} | {error, {term(), term()}}.
+send_tx(Tx) ->
```

Comment:
Why so low level? Imo we should provide interface just to publish a commitment, we don't need to process arbitrary txs